src/components/AutoCompleteInput/AutoCompleteInput.js

```diff
@@ -1,258 +1,232 @@
-import React from 'react';
+import React, { useContext, useEffect, useRef, useState } from 'react';
 import styled from '@stream-io/styled-components';
 import PropTypes from 'prop-types';
 
-import { withTranslationContext } from '../../context';
+import { SuggestionsContext, TranslationContext } from '../../context';
 
 const InputBox = styled.TextInput`
-  max-height: 60px;
-  margin: -5px;
   flex: 1;
+  margin: -5px;
+  max-height: 60px;
   ${({ theme }) => theme.messageInput.inputBox.css}
 `;
 
-class AutoCompleteInput extends React.PureComponent {
-  static propTypes = {
-    value: PropTypes.string,
-    /** @see See [suggestions context](https://getstream.github.io/stream-chat-react-native/#suggestionscontext) */
-    openSuggestions: PropTypes.func,
-    /** @see See [suggestions context](https://getstream.github.io/stream-chat-react-native/#suggestionscontext) */
-    closeSuggestions: PropTypes.func,
-    /** @see See [suggestions context](https://getstream.github.io/stream-chat-react-native/#suggestionscontext) */
-    updateSuggestions: PropTypes.func,
-    triggerSettings: PropTypes.object,
-    setInputBoxRef: PropTypes.func,
-    /**
-     * @param text string
-     */
-    onChange: PropTypes.func,
-    /**
-     * Additional props for underlying TextInput component. These props will be forwarded as it is to TextInput component.
-     *
-     * @see See https://facebook.github.io/react-native/docs/textinput#reference
-     */
-    additionalTextInputProps: PropTypes.object,
-  };
-
-  static defaultProps = {
-    value: undefined,
-  };
-
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      selectionStart: 0,
-      selectionEnd: 0,
-      currentTrigger: null,
-    };
+const computeCaretPosition = (token, startOfTokenPosition) =>
+  startOfTokenPosition + token.length;
 
-    this.isTrackingStarted = false;
+const isCommand = (text) => {
+  if (text[0] !== '/') {
+    return false;
   }
 
-  componentDidUpdate(prevProps) {
-    if (prevProps.value !== this.props.value) {
-      this.handleChange(this.props.value, true);
-    }
+  const tokens = text.split(' ');
+
+  if (tokens.length > 1) {
+    return false;
   }
 
-  startTracking = () => {
-    this.isTrackingStarted = true;
-    const { component, title } = this.props.triggerSettings[
-      this.state.currentTrigger
-    ];
-    this.props.openSuggestions(title, component);
+  return true;
+};
+
+const AutoCompleteInput = ({
+  additionalTextInputProps,
+  onChange,
+  setInputBoxRef,
+  triggerSettings,
+  value,
+}) => {
+  const {
+    closeSuggestions,
+    openSuggestions,
+    updateSuggestions: updateSuggestionsContext,
+  } = useContext(SuggestionsContext);
+  const { t } = useContext(TranslationContext);
+
+  const isTrackingStarted = useRef(false);
+
+  const [selectionEnd, setSelectionEnd] = useState(0);
+
+  useEffect(() => {
+    handleChange(value, true);
+  }, [handleChange, value]);
```

Comment:
does handleChange need to be in the dep array?

Reply:
doesn't hurt